### ISSUE TITLE
Server Does not Appear to Work

### ISSUE THREAD

**Issue by jherdman** [rubocop/rubocop#10842]
After reading an article about the new Rubocop server I eagerly attempted it on a project that uses Nix to manage the dev environment. I believe something is awry, possibly related to #10771 (wild guess).

--------

## Expected behavior

Initializing a Rubocop server works, and continues to work upon subsequent calls to Rubocop.

## Actual behavior

Sample shell commands from a session:

```shell
$ bin/rubocop --start-server
RuboCop server starting on 127.0.0.1:61108.

$ bin/rubocop --server-status
RuboCop server (16113) is running.

$ time bin/rubocop
2607 files inspected, no offenses detected

real	0m23.144s
user	0m0.400s
sys	0m0.107s

$ bin/rubocop --server-status
RuboCop server is not running.
```

### `rubocop --debug`

```
For /Users/james/Projects/PN/eternal-sledgehammer: configuration from /Users/james/Projects/PN/eternal-sledgehammer/.rubocop.yml
configuration from /Users/james/Projects/PN/eternal-sledgehammer/vendor/cache/ruby/2.7.0/gems/rubocop-performance-1.14.3/config/default.yml
configuration from /Users/james/Projects/PN/eternal-sledgehammer/vendor/cache/ruby/2.7.0/gems/rubocop-performance-1.14.3/config/default.yml
Default configuration from /Users/james/Projects/PN/eternal-sledgehammer/vendor/cache/ruby/2.7.0/gems/rubocop-1.32.0/config/default.yml
configuration from /Users/james/Projects/PN/eternal-sledgehammer/vendor/cache/ruby/2.7.0/gems/rubocop-rails-2.15.2/config/default.yml
configuration from /Users/james/Projects/PN/eternal-sledgehammer/vendor/cache/ruby/2.7.0/gems/rubocop-rails-2.15.2/config/default.yml
configuration from /Users/james/Projects/PN/eternal-sledgehammer/vendor/cache/ruby/2.7.0/gems/rubocop-rspec-2.12.1/config/default.yml
configuration from /Users/james/Projects/PN/eternal-sledgehammer/vendor/cache/ruby/2.7.0/gems/rubocop-rspec-2.12.1/config/default.yml
Inheriting configuration from /Users/james/Projects/PN/eternal-sledgehammer/.rubocop_todo.yml
.rubocop.yml: Metrics/MethodLength:Exclude overrides the same parameter in .rubocop_todo.yml
Use parallel by default.
Running parallel inspection

# elide indications of cache scanning
```

Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

## Steps to reproduce the problem

See above

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.32.0 (using Parser 3.1.2.0, rubocop-ast 1.19.1, running on ruby 2.7.6 arm64-darwin21)
  - rubocop-performance 1.14.3
  - rubocop-rails 2.15.2
  - rubocop-rspec 2.12.1
```
